homeassistant/components/hassio/jobs.py

```diff
@@ -158,3 +166,14 @@ def _process_job_change(self, job: Job) -> None:
         for sub in self._subscriptions:
             if sub.matches(job):
                 sub.event_callback(job)
+
+        # If the job is done, pop it from our cache if present after processing is done
+        if job.done and job.uuid in self._jobs:
+            del self._jobs[job.uuid]
```

Comment:
@MartinHjelmare I decided to add a couple tests of the new class along with the one you suggested and I'm glad I did. I realized there was another memory leak here, done jobs would build up. Removing done jobs from the cache after we have finished processing the update clears that up.